src/calendar/index.tsx

```diff
@@ -280,7 +296,7 @@ class Calendar extends Component<CalendarProps, CalendarState> {
 
   renderHeader() {
     const {customHeader, headerStyle, displayLoadingIndicator, markedDates, testID} = this.props;
-    const current = parseDate(this.props.current);
+    const current = parseDate(this.currentDate);
```

Comment:
Why are you talking the current date from the `this` and not from the state? 
In general, why are we keeping the current date also on `this`?

Reply:
Changed